File: php/Memory.php
<?php

// import file kelas parent
include "Hardware.php";

// deklarasi kelas Memory dengan inheritance terhadap kelas Hardware
class Memory extends Hardware{

    // deklarasi dan inisialisasi atribut private
    private $frequency = 0;
    private $memorySize = 0;
    private $supportCuda = "-";
    

    // constructor dengan parameter
    public function __construct($inputFreq = 0, $inputSize = 0, $inputSupCuda = ""){
        $this->frequency = $inputFreq;
        $this->memorySize = $inputSize;
        $this->supportCuda = $inputSupCuda;
    }


    // deklarasi setter sebagai method (Write)
    public function setFreq($freq){
        $this->frequency = $freq;
    }

    public function setSize($size){
        $this->memorySize = $size;
    }

    public function setSupCuda($supCuda){
        $this->supportCuda = $supCuda;
    }

    
    // deklarasi getter sebagai method (Read Only)
    public function getFreq(){
        return $this->frequency;
    }

    public function getSize(){
        return $this->memorySize;
    }

    public function getSupCuda(){
        return $this->supportCuda;
    }

    
    // deklarasi method Display untuk menampilkan data menggunakan method Getter
    public function displayMemory(){
        echo "Frequency    : ". $this->getFreq(). " MHz". "<br>";
        echo "Memory Size  : ". $this->getSize(). " GB". "<br>";
        echo "Support Cuda : ". $this->getSupCuda(). "<br>";
    }

    // destructor
    public function __destruct(){

    }
}

?>

File: python/Memory.py

# import file kelas parent
from Hardware import Hardware

# deklarasi kelas Memory dengan inheritance terhadap kelas Hardware
class Memory(Hardware):
    
    # deklarasi dan inisialisasi atribut private
    __frequency = 0
    __memorySize = 0
    __supportCuda = "-"        


    # constructor dengan parameter
    def __init__(self, freq = 0, size = 0, supCuda = ""):
        
        self.__frequency = freq
        self.__memorySize = size
        self.__supportCuda = supCuda

    
    # deklarasi setter sebagai method (Write)
    def setFreq(self, freq):
        self.__frequency = freq
    
    def setSize(self, size):
        self.__memorySize = size
    
    def setSupCuda(self, supCuda):
        self.__supportCuda = supCuda

    
    # deklarasi getter sebagai method (Read Only)
    def getFreq(self):
        return self.__frequency
    
    def getSize(self):
        return self.__memorySize
    
    def getSupCuda(self):
        return self.__supportCuda

    
    # deklarasi method Display untuk menampilkan data menggunakan method Getter
    def displayMemory(self):
        print("\nFrequency    : " + str(self.getFreq()) + " MHz")
        print("Memory Size  : " + str(self.getSize()) + " GB")
        print("Support Cuda : " + str(self.getSupCuda()))